gcc5 failed to install under Tiger / iMac G3

Hi!
I just tried to install gcc5 with brew, as:

`brew install gcc5`

and also with:

`brew install gcc5 --cc=gcc-4.2`

with the following output / error:

```
==> Downloading https://ftpmirror.gnu.org/gcc/gcc-5.4.0/gcc-5.4.0.tar.bz2
Already downloaded: /Library/Caches/Homebrew/gcc5-5.4.0.tar.bz2
==> Patching
patching file gcc/jit/Make-lang.in
==> ../configure --build=powerpc-apple-darwin8.11.0 --prefix=/usr/local/Cellar/gcc5/5.4.0 --libdir=/usr/local/Cellar/gcc5/5.4.0/lib/gcc/5 --enab
==> make bootstrap

Last 15 lines from /Users/manuelmc/Library/Logs/Homebrew/gcc5/02.make:
/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/./prev-gcc/xg++ -B/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/./prev-gcc/ -B/usr/local/Cellar/gcc5/5.4.0/powerpc-apple-darwin8.11.0/bin/ -nostdinc++ -B/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/src/.libs -B/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/libsupc++/.libs  -I/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/include/powerpc-apple-darwin8.11.0  -I/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/include  -I/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/libstdc++-v3/libsupc++ -L/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/src/.libs -L/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/libsupc++/.libs -c   -g -O2 -mdynamic-no-pic  -gtoggle -DIN_GCC    -fno-exceptions -fno-rtti -fasynchronous-unwind-tables -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wmissing-format-attribute -Woverloaded-virtual -pedantic -Wno-long-long -Wno-variadic-macros -Wno-overlength-strings   -DHAVE_CONFIG_H -I. -I. -I../../gcc -I../../gcc/. -I../../gcc/../include -I../../gcc/../libcpp/include -I/usr/local/opt/gmp/include -I/usr/local/opt/mpfr/include -I/usr/local/opt/libmpc/include  -I../../gcc/../libdecnumber -I../../gcc/../libdecnumber/dpd -I../libdecnumber -I../../gcc/../libbacktrace -I/usr/local/opt/isl014/include -I/usr/local/opt/cctools/include -I/usr/local/opt/isl014/include -F/usr/local/Frameworks -o gimple-fold.o -MT gimple-fold.o -MMD -MP -MF ./.deps/gimple-fold.TPo ../../gcc/gimple-fold.c
/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/./prev-gcc/xg++ -B/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/./prev-gcc/ -B/usr/local/Cellar/gcc5/5.4.0/powerpc-apple-darwin8.11.0/bin/ -nostdinc++ -B/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/src/.libs -B/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/libsupc++/.libs  -I/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/include/powerpc-apple-darwin8.11.0  -I/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/include  -I/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/libstdc++-v3/libsupc++ -L/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/src/.libs -L/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/libsupc++/.libs -c   -g -O2 -mdynamic-no-pic  -gtoggle -DIN_GCC    -fno-exceptions -fno-rtti -fasynchronous-unwind-tables -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wmissing-format-attribute -Woverloaded-virtual -pedantic -Wno-long-long -Wno-variadic-macros -Wno-overlength-strings   -DHAVE_CONFIG_H -I. -I. -I../../gcc -I../../gcc/. -I../../gcc/../include -I../../gcc/../libcpp/include -I/usr/local/opt/gmp/include -I/usr/local/opt/mpfr/include -I/usr/local/opt/libmpc/include  -I../../gcc/../libdecnumber -I../../gcc/../libdecnumber/dpd -I../libdecnumber -I../../gcc/../libbacktrace -I/usr/local/opt/isl014/include -I/usr/local/opt/cctools/include -I/usr/local/opt/isl014/include -F/usr/local/Frameworks -o gimple-low.o -MT gimple-low.o -MMD -MP -MF ./.deps/gimple-low.TPo ../../gcc/gimple-low.c
/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/./prev-gcc/xg++ -B/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/./prev-gcc/ -B/usr/local/Cellar/gcc5/5.4.0/powerpc-apple-darwin8.11.0/bin/ -nostdinc++ -B/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/src/.libs -B/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/libsupc++/.libs  -I/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/include/powerpc-apple-darwin8.11.0  -I/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/include  -I/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/libstdc++-v3/libsupc++ -L/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/src/.libs -L/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/libsupc++/.libs -c   -g -O2 -mdynamic-no-pic  -gtoggle -DIN_GCC    -fno-exceptions -fno-rtti -fasynchronous-unwind-tables -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wmissing-format-attribute -Woverloaded-virtual -pedantic -Wno-long-long -Wno-variadic-macros -Wno-overlength-strings   -DHAVE_CONFIG_H -DGENERATOR_FILE -I. -Ibuild -I../../gcc -I../../gcc/build -I../../gcc/../include  -I../../gcc/../libcpp/include -I/usr/local/opt/cctools/include -I/usr/local/opt/isl014/include -F/usr/local/Frameworks \
        -o build/genmatch.o ../../gcc/genmatch.c
/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/./prev-gcc/xg++ -B/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/./prev-gcc/ -B/usr/local/Cellar/gcc5/5.4.0/powerpc-apple-darwin8.11.0/bin/ -nostdinc++ -B/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/src/.libs -B/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/libsupc++/.libs  -I/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/include/powerpc-apple-darwin8.11.0  -I/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/include  -I/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/libstdc++-v3/libsupc++ -L/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/src/.libs -L/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/libsupc++/.libs -c   -g -O2 -mdynamic-no-pic  -gtoggle -DIN_GCC    -fno-exceptions -fno-rtti -fasynchronous-unwind-tables -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wmissing-format-attribute -Woverloaded-virtual -pedantic -Wno-long-long -Wno-variadic-macros -Wno-overlength-strings   -DHAVE_CONFIG_H -DGENERATOR_FILE -I. -Ibuild -I../../gcc -I../../gcc/build -I../../gcc/../include  -I../../gcc/../libcpp/include -I/usr/local/opt/cctools/include -I/usr/local/opt/isl014/include -F/usr/local/Frameworks \
        -o build/hash-table.o ../../gcc/hash-table.c
/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/./prev-gcc/xg++ -B/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/./prev-gcc/ -B/usr/local/Cellar/gcc5/5.4.0/powerpc-apple-darwin8.11.0/bin/ -nostdinc++ -B/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/src/.libs -B/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/libsupc++/.libs  -I/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/include/powerpc-apple-darwin8.11.0  -I/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/include  -I/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/libstdc++-v3/libsupc++ -L/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/src/.libs -L/private/tmp/gcc520171118-458-1b9p5yw/gcc-5.4.0/build/prev-powerpc-apple-darwin8.11.0/libstdc++-v3/libsupc++/.libs   -g -O2 -mdynamic-no-pic  -gtoggle -DIN_GCC    -fno-exceptions -fno-rtti -fasynchronous-unwind-tables -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wmissing-format-attribute -Woverloaded-virtual -pedantic -Wno-long-long -Wno-variadic-macros -Wno-overlength-strings   -DHAVE_CONFIG_H -DGENERATOR_FILE -static-libstdc++ -static-libgcc  -o build/genmatch \
    build/genmatch.o ../libcpp/libcpp.a ../libiberty/libiberty.a build/errors.o build/vec.o build/hash-table.o .././libiberty/libiberty.a  -liconv
build/genmatch --gimple ../../gcc/match.pd \
    > tmp-gimple-match.c
dyld: incompatible cpu-subtype
make[3]: *** [s-match] Error 133
make[2]: *** [all-stage2-gcc] Error 2
make[1]: *** [stage2-bubble] Error 2
make: *** [bootstrap] Error 2

READ THIS: https://github.com/mistydemeo/tigerbrew/wiki/troubleshooting
```

My configuration is the following (iMac G3, Mac OS Tiger 10.4.11 with all patches):

```
$ brew --config
HOMEBREW_VERSION: 0.9.5
ORIGIN: https://github.com/mistydemeo/tigerbrew.git
HEAD: 405110641660fb4e6aee3dfcbdf3484b4f0497e4
Last commit: 5 days ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://ia902307.us.archive.org/31/items/tigerbrew
CPU: single-core 32-bit g3
OS X: 10.4.11-Power Macintosh
Xcode: 2.5
CLT: N/A
GCC-4.0: build 5370
GCC-4.2: build 5553
Clang: N/A
X11: 1.1.3 => /usr/X11R6
System Ruby: 1.8.2 => /usr/bin/ruby
Perl: /usr/bin/perl
Python: /usr/local/bin/python => /usr/local/Cellar/python/2.7.12_3/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/bin/ruby => /Users/manuelmc
Java: N/A
```

Sadly, I was not able to execute 'gist-logs' to attach it to the issue:

```
$ brew gist-logs gcc5
Error: undefined method `ord' for 226:Fixnum
Please report this bug:
    https://github.com/mistydemeo/tigerbrew/wiki/troubleshooting
/usr/local/Library/Homebrew/vendor/okjson.rb:506:in `ucharcopy'
/usr/local/Library/Homebrew/vendor/okjson.rb:478:in `strenc'
/usr/local/Library/Homebrew/vendor/okjson.rb:77:in `valenc'
/usr/local/Library/Homebrew/vendor/okjson.rb:428:in `objenc'
/usr/local/Library/Homebrew/vendor/okjson.rb:428:in `map'
/usr/local/Library/Homebrew/vendor/okjson.rb:428:in `each'
/usr/local/Library/Homebrew/vendor/okjson.rb:428:in `map'
/usr/local/Library/Homebrew/vendor/okjson.rb:428:in `objenc'
/usr/local/Library/Homebrew/vendor/okjson.rb:75:in `valenc'
/usr/local/Library/Homebrew/vendor/okjson.rb:428:in `objenc'
/usr/local/Library/Homebrew/vendor/okjson.rb:428:in `map'
/usr/local/Library/Homebrew/vendor/okjson.rb:428:in `each'
/usr/local/Library/Homebrew/vendor/okjson.rb:428:in `map'
/usr/local/Library/Homebrew/vendor/okjson.rb:428:in `objenc'
/usr/local/Library/Homebrew/vendor/okjson.rb:75:in `valenc'
/usr/local/Library/Homebrew/vendor/okjson.rb:428:in `objenc'
/usr/local/Library/Homebrew/vendor/okjson.rb:428:in `map'
/usr/local/Library/Homebrew/vendor/okjson.rb:428:in `each'
/usr/local/Library/Homebrew/vendor/okjson.rb:428:in `map'
/usr/local/Library/Homebrew/vendor/okjson.rb:428:in `objenc'
/usr/local/Library/Homebrew/vendor/okjson.rb:65:in `encode'
/usr/local/Library/Homebrew/utils/json.rb:16:in `dump'
/usr/local/Library/Homebrew/cmd/gist-logs.rb:108:in `make_request'
/usr/local/Library/Homebrew/cmd/gist-logs.rb:113:in `post'
/usr/local/Library/Homebrew/cmd/gist-logs.rb:72:in `create_gist'
/usr/local/Library/Homebrew/cmd/gist-logs.rb:24:in `gistify_logs'
/usr/local/Library/Homebrew/cmd/gist-logs.rb:140:in `gist_logs'
/usr/local/Library/brew.rb:143:in `send'
/usr/local/Library/brew.rb:143
```

Regards,

```
build/genmatch --gimple ../../gcc/match.pd \
    > tmp-gimple-match.c
dyld: incompatible cpu-subtype
make[3]: *** [s-match] Error 133
make[2]: *** [all-stage2-gcc] Error 2
make[1]: *** [stage2-bubble] Error 2
make: *** [bootstrap] Error 2
```

Hmmm, that's interesting... I wonder if it's related to your being on a G3? Would you be able to upload the files from `~/Library/Logs/Homebrew/gcc` please?

--- a/Library/Formula/gcc44.rb
+++ b/Library/Formula/gcc44.rb
@@ -68,6 +68,10 @@ class Gcc44 < Formula
     # GCC will suffer build errors if forced to use a particular linker.
     ENV.delete "LD"
 
+    # Otherwise libstdc++ will be incorrectly tagged with cpusubtype 10 (G4e)
+    # https://github.com/mistydemeo/tigerbrew/issues/538
+    ENV.append_to_cflags "-force_cpusubtype_ALL" if Hardware::CPU.family == :g3
+
     if build.with? "all-languages"
       # Everything but Ada, which requires a pre-existing GCC Ada compiler
       # (gnat) to bootstrap.

--- a/Library/Formula/gcc45.rb
+++ b/Library/Formula/gcc45.rb
@@ -85,6 +85,10 @@ class Gcc45 < Formula
     # GCC will suffer build errors if forced to use a particular linker.
     ENV.delete "LD"
 
+    # Otherwise libstdc++ will be incorrectly tagged with cpusubtype 10 (G4e)
+    # https://github.com/mistydemeo/tigerbrew/issues/538
+    ENV.append_to_cflags "-force_cpusubtype_ALL" if Hardware::CPU.family == :g3
+
     if build.with? "all-languages"
       # Everything but Ada, which requires a pre-existing GCC Ada compiler
       # (gnat) to bootstrap.

--- a/Library/Formula/gcc46.rb
+++ b/Library/Formula/gcc46.rb
@@ -81,6 +81,10 @@ class Gcc46 < Formula
     # GCC will suffer build errors if forced to use a particular linker.
     ENV.delete "LD"
 
+    # Otherwise libstdc++ will be incorrectly tagged with cpusubtype 10 (G4e)
+    # https://github.com/mistydemeo/tigerbrew/issues/538
+    ENV.append_to_cflags "-force_cpusubtype_ALL" if Hardware::CPU.family == :g3
+
     if build.with? "all-languages"
       # Everything but Ada, which requires a pre-existing GCC Ada compiler
       # (gnat) to bootstrap. GCC 4.6.0 add go as a language option, but it is

--- a/Library/Formula/gcc47.rb
+++ b/Library/Formula/gcc47.rb
@@ -83,6 +83,10 @@ class Gcc47 < Formula
     # GCC will suffer build errors if forced to use a particular linker.
     ENV.delete "LD"
 
+    # Otherwise libstdc++ will be incorrectly tagged with cpusubtype 10 (G4e)
+    # https://github.com/mistydemeo/tigerbrew/issues/538
+    ENV.append_to_cflags "-force_cpusubtype_ALL" if Hardware::CPU.family == :g3
+
     if build.with? "all-languages"
       # Everything but Ada, which requires a pre-existing GCC Ada compiler
       # (gnat) to bootstrap. GCC 4.6.0 add go as a language option, but it is

--- a/Library/Formula/gcc48.rb
+++ b/Library/Formula/gcc48.rb
@@ -89,6 +89,10 @@ class Gcc48 < Formula
     # GCC will suffer build errors if forced to use a particular linker.
     ENV.delete "LD"
 
+    # Otherwise libstdc++ will be incorrectly tagged with cpusubtype 10 (G4e)
+    # https://github.com/mistydemeo/tigerbrew/issues/538
+    ENV.append_to_cflags "-force_cpusubtype_ALL" if Hardware::CPU.family == :g3
+
     if MacOS.version < :leopard
       ENV["AS"] = ENV["AS_FOR_TARGET"] = "#{Formula["cctools"].bin}/as"
     end

--- a/Library/Formula/gcc49.rb
+++ b/Library/Formula/gcc49.rb
@@ -83,6 +83,10 @@ class Gcc49 < Formula
     # GCC will suffer build errors if forced to use a particular linker.
     ENV.delete "LD"
 
+    # Otherwise libstdc++ will be incorrectly tagged with cpusubtype 10 (G4e)
+    # https://github.com/mistydemeo/tigerbrew/issues/538
+    ENV.append_to_cflags "-force_cpusubtype_ALL" if Hardware::CPU.family == :g3
+
     if build.with? "all-languages"
       # Everything but Ada, which requires a pre-existing GCC Ada compiler
       # (gnat) to bootstrap. GCC 4.6.0 add go as a language option, but it is

--- a/Library/Formula/gcc5.rb
+++ b/Library/Formula/gcc5.rb
@@ -71,6 +71,10 @@ class Gcc5 < Formula
     # GCC will suffer build errors if forced to use a particular linker.
     ENV.delete "LD"
 
+    # Otherwise libstdc++ will be incorrectly tagged with cpusubtype 10 (G4e)
+    # https://github.com/mistydemeo/tigerbrew/issues/538
+    ENV.append_to_cflags "-force_cpusubtype_ALL" if Hardware::CPU.family == :g3
+
     if MacOS.version < :leopard
       ENV["AS"] = ENV["AS_FOR_TARGET"] = "#{Formula["cctools"].bin}/as"
     end

--- a/Library/Formula/gcc6.rb
+++ b/Library/Formula/gcc6.rb
@@ -71,6 +71,10 @@ class Gcc6 < Formula
     # GCC will suffer build errors if forced to use a particular linker.
     ENV.delete "LD"
 
+    # Otherwise libstdc++ will be incorrectly tagged with cpusubtype 10 (G4e)
+    # https://github.com/mistydemeo/tigerbrew/issues/538
+    ENV.append_to_cflags "-force_cpusubtype_ALL" if Hardware::CPU.family == :g3
+
     if MacOS.version < :leopard
       ENV["AS"] = ENV["AS_FOR_TARGET"] = "#{Formula["cctools"].bin}/as"
     end

